client/input.go

```diff
@@ -28,12 +28,17 @@ func GetPassword(prompt string, buf *bufio.Reader) (pass string, err error) {
 	} else {
 		pass, err = readLineFromBuf(buf)
 	}
+
 	if err != nil {
 		return "", err
 	}
+
 	if len(pass) < MinPassLength {
-		return "", errors.Errorf("password must be at least %d characters", MinPassLength)
+		// Return the given password to the upstream client so it can handle a
+		// non-STDIN failure gracefully.
```

Comment:
typically we try to keep comments as non-verbose as possible, which often means they are not sentences - I think this makes sense - here just a general comment